src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs

```diff
@@ -2541,48 +2528,47 @@ public async Task TestOpenProject_CrossLanguageSkeletonReferenceHasDocComments()
             Assert.Equal(cscomment, vbcomment);
         }
 
-        [ConditionalFact(typeof(VisualStudioMSBuildInstalled)), Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
-        public void TestOpenProject_WithProjectFileLocked()
+        [ConditionalFact(typeof(VisualStudioMSBuildInstalled), typeof(IsEnglishLocal)), Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
+        public async Task TestOpenProject_WithProjectFileLockedAsync()
         {
             CreateFiles(GetSimpleCSharpSolutionFiles());
 
             // open for read-write so no-one else can read
             var projectFile = GetSolutionFileName(@"CSharpProject\CSharpProject.csproj");
             using (File.Open(projectFile, FileMode.Open, FileAccess.ReadWrite))
             {
-                AssertEx.Throws<IOException>(() =>
-                    {
-                        using var workspace = CreateMSBuildWorkspace();
-                        workspace.OpenProjectAsync(projectFile).Wait();
-                    });
+                using var workspace = CreateMSBuildWorkspace();
+                await workspace.OpenProjectAsync(projectFile);
+                var diagnostic = Assert.Single(workspace.Diagnostics);
+                Assert.Contains("The process cannot access the file", diagnostic.Message);
```

Comment:
This is asserting what the code actually does. It didn't throw an exception, and since AssertEx.Throws was bugged, we never saw.